pulp_ostree/tests/functional/api/test_filter.py

```diff
@@ -1,100 +1,69 @@
-        remote = self.remotes_api.create(body)
-        self.addCleanup(self.remotes_api.delete, remote.pulp_href)
-
-        self.assertEqual(self.repo.latest_version_href, f"{self.repo.pulp_href}versions/0/")
-        repository_sync_data = RepositorySyncURL(remote=remote.pulp_href)
-        response = self.repositories_api.sync(self.repo.pulp_href, repository_sync_data)
-        repo_version = monitor_task(response.task).created_resources[0]
-
-        self.repository_version = self.versions_api.read(repo_version)
-
-    def test_filter_refs(self):
-        """Check if refs can be filtered by their names and related commits' checksums."""
-        refs_stable = self.refs_api.list(
-            repository_version_added=self.repository_version.pulp_href, name="stable"
-        ).to_dict()
-
-        self.assertEqual(refs_stable["count"], 1, refs_stable)
-        self.assertEqual(refs_stable["results"][0]["name"], "stable", refs_stable)
-
-        commit = self.commits_api.read(refs_stable["results"][0]["commit"])
-
-        refs_commit_checksum = self.refs_api.list(
-            repository_version_added=self.repository_version.pulp_href, checksum=commit.checksum
-        ).to_dict()
-
-        self.assertEqual(refs_commit_checksum["count"], 1, refs_commit_checksum)
-        self.assertEqual(
-            refs_commit_checksum["results"][0]["checksum"], commit.checksum, refs_commit_checksum
-        )
-
-    def test_filter_commits(self):
-        """Check if commits can be filtered by their checksums."""
-        refs_rawhide = self.refs_api.list(
-            repository_version_added=self.repository_version.pulp_href, name="rawhide"
-        ).to_dict()
-        ref_rawhide = refs_rawhide["results"][0]
-
-        commits_rawhide = self.commits_api.list(
-            repository_version_added=self.repository_version.pulp_href,
-            checksum=ref_rawhide["checksum"],
-        ).to_dict()
-        self.assertEqual(commits_rawhide["count"], 1, commits_rawhide)
-
-        commit_rawhide = commits_rawhide["results"][0]
-        self.assertEqual(commit_rawhide["checksum"], ref_rawhide["checksum"], commit_rawhide)
-        self.assertEqual(commit_rawhide["pulp_href"], ref_rawhide["commit"], commit_rawhide)
+import pytest
+
+from pulp_ostree.tests.functional.constants import OSTREE_FIXTURE_URL
+
+
+@pytest.fixture
+def synced_repo(
+    ostree_repositories_api_client, ostree_repository_factory, ostree_remote_factory, monitor_task
+):
+    """
+    Fixture that syncs a Remote ostree Repository,
+    waits until the sync task completes and returns the
+    created repository object.
+    """
+
+    def _synced_repo():
+        repo = ostree_repository_factory()
+        remote = ostree_remote_factory(url=OSTREE_FIXTURE_URL, policy="immediate")
+        result = ostree_repositories_api_client.sync(repo.pulp_href, {"remote": remote.pulp_href})
+        monitor_task(result.task)
+        return ostree_repositories_api_client.read(repo.pulp_href)
+
```

Comment:
There is no need for having an inner function anymore. We can simply return `ostree_repositories_api_client.read(repo.pulp_href)` from `synced_repo`.